src/shared/sam/localLambdaRunner.ts

```diff
@@ -465,65 +465,52 @@ vscode.debug.onDidTerminateDebugSession(session => {
  * Sends an HTTP request to the local API webserver, which invokes the backing
  * Lambda, which will then enter debugging.
  */
-function requestLocalApi(ctx: ExtContext, api: APIGatewayProperties, apiPort: number, payload: any): Promise<void> {
-    return new Promise((resolve, reject) => {
-        const reqMethod = api?.httpMethod?.toUpperCase() ?? 'GET'
-        const qs = (api?.querystring?.startsWith('?') ? '' : '?') + (api?.querystring ?? '')
-        const reqOpts = {
-            // Sets body to JSON value and adds Content-type: application/json header.
-            json: true,
-            uri: `http://127.0.0.1:${apiPort}${api?.path}${qs}`,
-            method: reqMethod,
-            timeout: 4000,
-            headers: api?.headers,
-            body: payload,
-            // TODO: api?.stageVariables,
-        }
-        getLogger('channel').info(
-            localize('AWS.sam.localApi.request', 'Sending request to local API: {0}', reqOpts.uri)
-        )
+async function requestLocalApi(
+    ctx: ExtContext,
+    api: APIGatewayProperties,
+    apiPort: number,
+    payload: any
+): Promise<void> {
+    const RETRY_LIMIT = 30
+    const RETRY_DELAY = 200
+    const reqMethod = api?.httpMethod || 'GET'
+    const qs = (api?.querystring?.startsWith('?') ? '' : '?') + (api?.querystring ?? '')
+    const uri = `http://127.0.0.1:${apiPort}${api?.path}${qs}`
+    const reqOpts: OptionsOfTextResponseBody = {
+        json: payload,
+        timeout: { socket: 1000 },
+        headers: api?.headers,
+        method: reqMethod,
+        retry: {
+            limit: RETRY_LIMIT,
+            statusCodes: [],
```

Comment:
I've made the retry behavior more granular in that it will never retry after receiving an invalid status code. Before it was only stopping after a `403`. Were there prior use-cases for retrying when the server is alive but responding with error codes?